### PR TITLE
cmake,common/RWLock: check for libpthread extensions

### PR DESCRIPTION
**Pull request by tchaikov**
pthread_rwlockattr_setkind_np() is a GNU extension of libpthread. and
Tianshan Qu pointed out, we cannot use
ifdef(PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP) to detect the
availability of this function, because it's an enum not a macro. so,
like other *_np() extensions, we check this one also using cmake at
the configure phase.

Reported-by: Tianshan Qu <tianshan@xsky.com>
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tianshan** [comment]
LTGM, seems more reasonable

File: CMakeLists.txt [hunk 1]
```diff
@@ -101,6 +101,7 @@ CHECK_FUNCTION_EXISTS(pthread_spin_init HAVE_PTHREAD_SPINLOCK)
 CHECK_FUNCTION_EXISTS(pthread_set_name_np HAVE_PTHREAD_SET_NAME_NP)
 CHECK_FUNCTION_EXISTS(pthread_setname_np HAVE_PTHREAD_SETNAME_NP)
 CHECK_FUNCTION_EXISTS(pthread_getname_np HAVE_PTHREAD_GETNAME_NP)
+CHECK_FUNCTION_EXISTS(pthread_rwlockattr_setkind_np HAVE_PTHREAD_RWLOCKATTR_SETKIND_NP)
```
Comment (cbodley) [inline CMakeLists.txt]: this sets a cmake variable `HAVE_PTHREAD_RWLOCKATTR_SETKIND_NP`

to make that visible to the compiler, we also need to add a `#cmakedefine` for it in `src/include/config-h.in.cmake`

Comment (tchaikov) [reply CMakeLists.txt]: @cbodley ahh, i forgot that! fixed and repushed.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-11-29_16:42:05-rados-wip-kefu-testing-2017-11-29-2317-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-30_07:07:30-rados-wip-kefu-testing-2017-11-29-2317-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2017-12-01_02:01:54-rados-wip-kefu-testing-2017-11-29-2317-distro-basic-mira/

**Pull request merged**